pkg/kotsadmsnapshot/backup.go

```diff
@@ -930,3 +949,68 @@ func prepareIncludedNamespaces(namespaces []string) []string {
 	}
 	return includedNamespaces
 }
+
+// excludeShutdownPodsFromBackup will exclude pods that are in a shutdown state from the backup
+// this is to prevent the hook backup from failing if a pod is in a shutdown state and cannot be backed up
+func excludeShutdownPodsFromBackup(ctx context.Context, clientset kubernetes.Interface, backupNamespaces []string, isKotsadmClusterScoped bool) (err error) {
+	failedPodListOptions := buildShutdownPodListOptions()
+
+	for _, namespace := range backupNamespaces {
+		if namespace == "*" {
+			if !isKotsadmClusterScoped {
+				continue
+			} else {
+				// if namespace is *, kubernetes api equivalent is empty string for all namespaces
+				namespace = ""
+			}
+		}
+
+		if err := excludeShutdownPodsFromBackupInNamespace(ctx, clientset, namespace, failedPodListOptions); err != nil {
+			return errors.Wrap(err, "failed to exclude shutdown pods from backup")
+		}
+	}
+
+	return nil
+}
+
+// excludeShutdownPodsFromBackupInNamespace will exclude pods that are in a shutdown state from the backup in a specific namespace
+func excludeShutdownPodsFromBackupInNamespace(ctx context.Context, clientset kubernetes.Interface, namespace string, failedPodListOptions metav1.ListOptions) error {
+	pods, err := clientset.CoreV1().Pods(namespace).List(ctx, failedPodListOptions)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list pods in namespace %s", namespace)
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase == corev1.PodFailed && pod.Status.Reason == "Shutdown" {
+			logger.Infof("Excluding pod %s in namespace %s from backup", pod.Name, namespace)
+			// add velero.io/exclude-from-backup=true label to pod
+			if pod.Labels == nil {
+				pod.Labels = map[string]string{}
+			}
+
+			pod.Labels[kotsadmtypes.ExcludeKey] = kotsadmtypes.ExcludeValue
+			_, err := clientset.CoreV1().Pods(pod.Namespace).Update(ctx, &pod, metav1.UpdateOptions{})
+			if err != nil {
+				return errors.Wrapf(err, "failed to update pod %s in namespace %s", pod.Name, pod.Namespace)
+			}
+		}
+	}
+	return nil
+}
+
+// buildShutdownPodListOptions returns a list options object that will match all pods that are in a failed state with shutdown reason
+func buildShutdownPodListOptions() metav1.ListOptions {
+	kotsadmLabelSet := labels.Set{
```

Comment:
those labels won't work for application resources. since selectors don't support "OR" operations for different keys, you'll probably need to list pods twice, once with the kotsadm key/value, and another with the app slug key/value.

Reply:
Don't the pods have both the labels?
I thought we were listing only pods with those 2 labels.

Reply:
kots components will have the `kots.io/kotsadm: "true"` label, and application pods will have the `kots.io/app-slug: <app-slug>` label

Reply:
made changes in https://github.com/replicatedhq/kots/pull/3610/commits/4f9e404dc195935e60ac9281f93baa7bef524d53